### ISSUE TITLE
undefined method not caught inside while loop

### ISSUE THREAD

**Issue by zanona** [vimeo/psalm#4147]
```php
<?php

foo(); //correctly caught as undefined

function bar(): void {
	while(true) {
		foo(); // should throw undefined method

                if (function_exists('any')) {} //method call plus below breaks check
                $b = 1; //defining variable plus above breaks check

                baz($b); // should throw undefined method
	}
}

```
https://psalm.dev/r/d3911e345a
